utils/src/main/kotlin/storage/FileStorage.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.utils.storage
+
+import java.io.InputStream
+
+/**
+ * A file storage backend, able to [read] and [write] files to a storage backend, like a local directory or a remote
+ * server.
+ */
+interface FileStorage {
+    /**
+     * Read the file at [path]. It is the callers responsibility to close the returned [InputStream] after consuming it.
+     */
+    fun read(path: String): InputStream
+
+    /**
+     * Write the data from [inputStream] to the file at [path]. If the file already exists it is overwritten. The
+     * provided [inputStream] is closed after writing it to the file.
+     */
+    fun write(path: String, inputStream: InputStream)
```

Comment:
Regarding the concurrency what do you think about the following cases
(1) read in parallel to write
(2) write in parallel to write

I think for (1) we can allow it ensuring that the read reads the outdate value, e.g. the read does not see the effect of the parallel write.
For (2) how this should be handled, in particular considering not only the remote http based case but also the local one. 

I wonder if the API is suitable in its  ```read``` part, in case a concurrent write happens.